testing/trino-testing/src/test/java/io/trino/testing/TestTestingTrinoClient.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.testing;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Key;
+import io.trino.Session;
+import io.trino.client.OkHttpUtil;
+import io.trino.execution.QueryIdGenerator;
+import io.trino.metadata.SessionPropertyManager;
+import io.trino.server.security.PasswordAuthenticatorManager;
+import io.trino.server.testing.TestingTrinoServer;
+import io.trino.spi.security.AccessDeniedException;
+import io.trino.spi.security.BasicPrincipal;
+import io.trino.spi.security.Identity;
+import okhttp3.OkHttpClient;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.security.Principal;
+
+import static org.testng.Assert.assertEquals;
+
+public class TestTestingTrinoClient
+{
+    private static final String TEST_USER = "test_user";
+    private static final String PASSWORD = "password";
+    private static final QueryIdGenerator queryIdGenerator = new QueryIdGenerator();
+    private static final SessionPropertyManager sessionManager = new SessionPropertyManager();
+    private static final Session session = Session.builder(sessionManager)
+            .setIdentity(Identity.forUser(TEST_USER).build())
+            .setQueryId(queryIdGenerator.createNextQueryId())
+            .build();
+
+    private TestingTrinoServer server;
+
+    @BeforeClass
+    public void setup()
+    {
+        server = TestingTrinoServer.builder()
+                .setProperties(ImmutableMap.<String, String>builder()
+                        .put("http-server.authentication.type", "password")
+                        .put("http-server.authentication.allow-insecure-over-http", "false")
+                        .put("http-server.process-forwarded", "true")
+                        .build())
+                .build();
+
+        server.getInstance(Key.get(PasswordAuthenticatorManager.class)).setAuthenticator(TestTestingTrinoClient::authenticate);
```

Comment:
this should be supported by `TestingTrinoServer` builder.

Reply:
I tried to support setting `Authenticator` in `TestingTrinoServer.Builder`, but it depends on the authentication type and authentication type is (and should be) configurable.

I couldn't easily find a way to support it in `TestingTrinoServer.Builder`.